CRM/MembershipExtras/Upgrader.php

```diff
@@ -26,14 +29,72 @@ private function createOfflineAutoRenewalScheduledJob() {
       'api_action' => 'run',
     ]);
 
+
     $paymentProcessorType = new ManualRecurringPaymentProcessorType();
     $paymentProcessorType->create();
 
     $paymentProcessor = new OfflineRecurringPaymentProcessor();
     $paymentProcessor->create();
   }
 
-  public function enable() {
+    /**
+     * Creates 'External ID' Custom Field for LineItem
+     */
+  private function createLineItemExternalIDCustomField()
+  {
+    $optionValues = civicrm_api3('OptionValue', 'get', [
+      'option_group_id' => 'cg_extend_objects',
+      'name' => 'civicrm_line_item'
+    ]);
+
+    if (!$optionValues['count']) {
+      civicrm_api3('OptionValue', 'create', [
+        'option_group_id' => 'cg_extend_objects',
+        'name' => 'civicrm_line_item',
+        'label' => ts('Line Item'),
+        'value' => 'LineItem',
+      ]);
+    }
+
+    $customGroups = civicrm_api3('CustomGroup', 'get', [
+      'extends' => 'LineItem',
+      'name' => 'recurring_contribution_external_id',
+    ]);
+
+    if (!$customGroups['count']) {
+      $customGroups = civicrm_api3('CustomGroup', 'create', [
+        'extends' => 'LineItem',
+        'name' => 'line_item_external_id',
+        'title' => E::ts('Line Item External ID'),
+        'table_name' => 'civicrm_value_line_item_ext_id',
+        'is_active' => 0,
+        'style' => 'Inline',
+        'is_multiple' => 0,
+
+      ]);
+    }
+
+    $customFields = civicrm_api3('CustomField', 'get', [
+      'custom_group_id' => $customGroups['id'],
+    ]);
```

Comment:
You should probably also check for the specific field, not just if the custom group has fields. Just in case.